### PR TITLE
modify_update() changes

### PR DESCRIPTION
**Pull request by ardbiesheuvel**

### CODE REVIEW

File: src/idmap.rs [hunk 1]
```diff
@@ -179,7 +179,7 @@ impl IdMap {
     /// largest virtual address covered by the page table given its root level.
```
Comment (jakobvukalovic) [inline src/idmap.rs:184]: Needs docs update - `walk_range` can also return `MapError::PteUpdateFault` now.

File: src/paging.rs [hunk 2]
```diff
@@ -604,21 +612,30 @@ impl<T: Translation> PageTableWithLevel<T> {
 
     /// Modifies a range of page table entries by applying a function to each page table entry.
     /// If the range is not aligned to block boundaries, it will be expanded.
```
Comment (qwandor) [inline src/paging.rs:614]: This is no longer true, right?

File: src/paging.rs [hunk 3]
```diff
@@ -711,8 +728,15 @@ impl Descriptor {
     }
 
     /// Modifies the page table entry by setting or clearing its flags.
+    /// Panicks when attempting to convert a table descriptor into a block/page descriptor or vice
```
Comment (qwandor) [inline src/paging.rs:731]: s/Panicks/Panics/

File: src/paging.rs (hunk 2 above)
Comment (ardbiesheuvel) [reply src/paging.rs:614]: indeed

File: src/paging.rs (hunk 3 above)
Comment (ardbiesheuvel) [reply src/paging.rs]: ok

File: src/idmap.rs (hunk 1 above)
Comment (ardbiesheuvel) [reply src/idmap.rs:186]: ack

**Review by qwandor — APPROVED** [review]
Looks good to me, as long as Jakob is happy that the old behaviour of expanding the range rather than splitting isn't needed anywhere.

**Pull request merged**